x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java

```diff
@@ -288,7 +282,7 @@ public LikePattern visitPattern(PatternContext ctx) {
             } else if (escapeString.length() == 1) {
                 escape = escapeString.charAt(0);
                 // these chars already have a meaning
-                if (escape == '*' || escape == '%' || escape == '_') {
+                if (escape == '%' || escape == '_') {
                     throw new ParsingException(source(escapeCtx.escape), "Char [{}] cannot be used for escaping", escape);
```

Comment:
On the same idea (which I like) of actually[ mentioning the wildcard chars in the error message itself](https://github.com/elastic/elasticsearch/pull/63428/files#diff-3c0e68f5f3ff479d4817a66493f2d7decb2974c2368cb68bf2bae612b2b91743R301) (for the lazy user who doesn't read the docs and asking "why % char cannot be used for escaping?"), what about adapting this error message to include the reason? Something like `Char [%] cannot be used for escaping as it's one of the wildcard chars [%_]`.